src/Tribe/Views/V2/Widgets/Widget_Abstract.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Tribe\Events\Views\V2\Widgets;
+
+use Tribe\Events\Views\V2\View;
+use Tribe\Events\Views\V2\View_Interface;
+use Tribe__Context as Context;
+
+/**
+ * The abstract all widgets should implement.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Widget
+ *
+ */
+abstract class Widget_Abstract extends \Tribe\Widget\Widget_Abstract {
+
+	/**
+	 * The view interface for the widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var View_Interface;
+	 */
+	protected $view;
+
+	/**
+	 * The slug of the widget view.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $view_slug;
+
+	/**
+	 * Setup the view for the widget.
+	 *
+	 * @since TBD
+	 */
+	public function setup_view() {
+		$context = tribe_context();
+
+		// Modifies the Context for the widget params.
+		// todo update per https://github.com/moderntribe/tribe-common/pull/1451#discussion_r501498990
+		$context = $this->alter_context( $context );
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 6ebe0e99b79b9295532f0fa30cf10572</sub>